data-document-model/src/main/java/io/micronaut/data/document/model/query/builder/CosmosSqlQueryBuilder.java

```diff
@@ -167,13 +179,175 @@ protected String getMappedName(NamingStrategy namingStrategy, List<Association>
             }
         }
         if (sb.length() > 0) {
-            sb.append(NameUtils.capitalize(property.getName()));
+            sb.append(DOT).append(property.getName());
         } else {
             sb.append(property.getName());
         }
         return namingStrategy.mappedName(sb.toString());
     }
 
+    @Override
+    protected void traversePersistentProperties(List<Association> associations,
+                                                PersistentProperty property,
+                                                boolean criteria,
+                                                BiConsumer<List<Association>, PersistentProperty> consumerProperty) {
+        if (property instanceof Embedded && !criteria) {
+            consumerProperty.accept(associations, property);
+            return;
+        }
+        super.traversePersistentProperties(associations, property, criteria, consumerProperty);
+    }
+
+    @Override
+    protected <T extends QueryModel.PropertyCriterion> CriterionHandler<T> inCriterionHandler(boolean negative) {
+        return (ctx, inQuery) -> {
+            QueryPropertyPath propertyPath = ctx.getRequiredProperty(inQuery.getProperty(), QueryModel.In.class);
+            StringBuilder whereClause = ctx.query();
+            if (negative) {
+                whereClause.append(" NOT ");
+            }
+            whereClause.append(" ARRAY_CONTAINS(");
```

Comment:
Had to implement it this way since parametrized IN criteria did not work, it is known issue:  https://github.com/Azure/azure-cosmosdb-node/issues/156

Reply:
Yes, that's better and I changed it. Tnx